src/simulate.py

```diff
@@ -801,6 +828,8 @@ def main(args):
     else:
         simulator.cache_size = 16
 
+    hw.gen_cacti_results()
```

Comment:
you can just call this inside `hardwareModel.__init__()` the simulator doesn't need to know about cacti's existence. 
This is where that encapsulation is useful.